Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -2002,31 +1997,17 @@
     CGContextStrokePath();
 
     // Lock the data that is backing the layer.
-    BYTE* buffer = static_cast<BYTE*>(_imgDest->Backing()->LockImageData());
-    THROW_NS_IF_NULL(E_UNEXPECTED, buffer);
-    int stride = _imgDest->Backing()->BytesPerRow();
-    float width = _imgDest->Backing()->Width();
+    _imgDest->Backing()->LockImageData();
```

Comment:
No the lock needs to be here so the renderTarget stays valid until its use has been completed. Ideally if we had used smart pointers all this would have come to us for free but I am going to keep CG code as is and rely on this single lock to keep the backing buffer alive

---
In reply to: [80771507](https://github.com/Microsoft/WinObjC/pull/1040#discussion_r80771507) [](ancestors = 80771507)